src/components/forms/CollisionLayerPicker.tsx

```diff
@@ -0,0 +1,188 @@
+import React, { useCallback, useMemo } from "react";
+import {useAppSelector } from "store/hooks";
+import styled from "styled-components";
+import { CollisionSetting } from "shared/lib/resources/types";
+import { COLLISIONS_EXTRA_SYMBOLS, defaultCollisionSettings } from "consts";
+import { 
+  BrushToolbarTileSolidIcon, 
+  BrushToolbarTileTopIcon, 
+  BrushToolbarTileBottomIcon, 
+  BrushToolbarTileLeftIcon, 
+  BrushToolbarTileRightIcon, 
+  BrushToolbarLadderTileIcon, 
+  BrushToolbarTileSlope45RightIcon, 
+  BrushToolbarTileSlope45LeftIcon, 
+  BrushToolbarTileSlope22RightBottomIcon, 
+  BrushToolbarTileSlope22RightTopIcon, 
+  BrushToolbarTileSlope22LeftTopIcon, 
+  BrushToolbarTileSlope22LeftBottomIcon, 
+  BrushToolbarExtraTileIcon 
+} from "components/world/BrushToolbarIcons";
+import { Input } from "ui/form/RenameInput";
+import l10n from "shared/lib/lang/l10n";
+
+const CollisionLayerStyle = styled.div`
+  display: flex;
+  width: 100%;
+  box-sizing: border-box;
+  column-gap: 15px;
+  padding: 2px;
+`
+
+interface CollisionColorProps {
+  $color: string;
+}
+
+const ColorInputStyle = styled.input<CollisionColorProps>`
+  margin-left: 8px;
+  width: 88px;
+  height: 32px;
+  background-color: ${(props) => props.$color};
+  color: black;
+  font-weight: bold;
+  font-family: monospace;
+  font-size: 15px;
+  autoFocus=true;
+`;
+
+type CollisionLayerPickerProps = {
+  onChange: (layer: CollisionSetting) => void;
+};
+
+export const CollisionLayerPicker = ({
+  onChange
+}: CollisionLayerPickerProps) => {
+
+  const settings = useAppSelector((state) => state.project.present.settings);
+  
+  const collisionSettings = useMemo(
+    () => settings.collisionSettings ?? defaultCollisionSettings, 
+    [settings.collisionSettings, defaultCollisionSettings]
+  );
+
+  return (<div> 
+    {collisionSettings.map((layer) => {
+      let icon;
+      let name = layer.name && layer.name.trim().length > 0 ? layer.name : undefined;
+      if (name) icon = <BrushToolbarExtraTileIcon $value={name[0]} $color={layer.color} />;
+      else { 
+        switch(layer.key) {
+          case "solid": 
+            name = l10n("FIELD_SOLID");
+            icon = <BrushToolbarTileSolidIcon  $color={layer.color} />; 
+            break;
+          case "top": 
+            name = l10n("FIELD_COLLISION_TOP");
+            icon = <BrushToolbarTileTopIcon $color={layer.color} />; 
+            break;
+          case "bottom": 
+            name = l10n("FIELD_COLLISION_BOTTOM");
+            icon = <BrushToolbarTileBottomIcon $color={layer.color} />; 
+            break;
+          case "left": 
+            name = l10n("FIELD_COLLISION_LEFT");
+            icon = <BrushToolbarTileLeftIcon $color={layer.color} />; 
+            break;
+          case "right": 
+            name = l10n("FIELD_COLLISION_RIGHT");
```

Comment:
There are a lot of these massive switch blocks because of how the bits of a tile are used (ie. the upper nibble is used as flags, but the lower nibble is used as an index). Perhaps some more options for how those bits can be used would be useful.